assets/js/app.js

```diff
@@ -0,0 +1,137 @@
+        countries.forEach((country) => {
+            const { name, population, region, languages } = country;
+
+            totalPopulation += population;
+
+            // Store languages per country
+            if (languages) {
+                countryLanguages[name.common] = Object.values(languages);
+                Object.values(languages).forEach((language) => {
+                    languagesCount[language] = (languagesCount[language] || 0) + 1;
+                });
+            }
+
+            const row = `
+                <tr>
+                    <td>${name.common}</td>
+                    <td>${population.toLocaleString()}</td>
+                    <td>${region}</td>
+                </tr>
+            `;
+            countriesTableBody.innerHTML += row;
+
+            // Count regions
+            regionCount[region] = (regionCount[region] || 0) + 1;
+        });
+
+        // Display global statistics
+        const avgPopulation = totalPopulation / countries.length;
+        statisticsDiv.innerHTML = `
+            <p>Total countries: ${countries.length}</p>
+            <p>Total population: ${totalPopulation.toLocaleString()}</p>
+            <p>Average population: ${avgPopulation.toFixed(2)}</p>
+        `;
+
+        // Display regions dynamically
+        Object.keys(regionCount).forEach((region) => {
+            const row = `
+                <tr>
+                    <td>${region}</td>
+                    <td>${regionCount[region]}</td>
+                </tr>
+            `;
+            regionTableBody.innerHTML += row;
+        });
+
+        // Display languages and their counts dynamically
+        Object.keys(languagesCount).forEach((language) => {
+            const row = `
+                <tr>
+                    <td>${language}</td>
+                    <td>${languagesCount[language]}</td>
+                </tr>
+            `;
+            languagesTableBody.innerHTML += row;
+        });
+
+        return { countryLanguages, languagesCount }; // Return the mapping of countries and their languages
+    };
+
+    // Function to fetch countries
+    const fetchCountries = (url) => {
+        return fetch(url)
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error("Failed to fetch data");
+                }
+                return response.json();
+            })
+            .catch((error) => {
+                console.error("Error:", error);
+                console.warn("Failed to load countries.");
+            });
+    };
+
+    // Button to display all countries
+    allButton.addEventListener("click", () => {
+        fetchCountries("https://restcountries.com/v3.1/all")
+            .then((countries) => {
+                displayCountries(countries);
+            })
+            .catch((error) => console.warn("Error fetching all countries:", error));
+    });
+
+    // Button to search for a country
+    searchButton.addEventListener("click", () => {
```

Comment:
this is the core problem: you should have had a single function used to display, and it should have got as param either all countries or some countries. but the logic for handling all countries or some countries should have been the same. You have implemented twice (not so DRY) so the app is inconsistent